src/test/java/com/teragrep/cfe_39/BatchDistributionTest.java

```diff
@@ -0,0 +1,684 @@
+                            .getBytes(StandardCharsets.UTF_8)
+            );
+            kafkaRecord = new KafkaRecordImpl(record.topic(), record.partition(), record.offset(), record.value());
+            kafkaRecordList.add(kafkaRecord);
+
+            output.accept(kafkaRecordList);
+
+            // Assert that records 11-13 are present in local avro-file.
+
+            File queueDirectory = new File(config.getQueueDirectory());
+            File[] files = queueDirectory.listFiles();
+            Assertions.assertEquals(1, files.length);
+
+            DatumReader<SyslogRecord> datumReader = new SpecificDatumReader<>(SyslogRecord.class);
+            DataFileReader<SyslogRecord> dataFileReader = new DataFileReader<>(files[0], datumReader);
+            Assertions.assertTrue(dataFileReader.hasNext());
+            SyslogRecord next = dataFileReader.next();
+            Assertions.assertEquals(11, next.getOffset());
+            Assertions.assertTrue(dataFileReader.hasNext());
+            next = dataFileReader.next();
+            Assertions.assertEquals(12, next.getOffset());
+            Assertions.assertTrue(dataFileReader.hasNext());
+            next = dataFileReader.next();
+            Assertions.assertEquals(13, next.getOffset());
+
+            // Assert that records 0-10 are present in HDFS
+
+            Assertions.assertEquals(1, fs.listStatus(new Path(config.getHdfsPath() + "/" + "topicName")).length);
+            Assertions.assertTrue(fs.exists(new Path(config.getHdfsPath() + "/" + "topicName" + "/" + "0.10")));
+            Path hdfsreadpath = new Path(config.getHdfsPath() + "/" + "topicName" + "/" + "0.10");
+            //Init input stream
+            FSDataInputStream inputStream = fs.open(hdfsreadpath);
+            //The data is in AVRO-format, so it can't be read as a string.
+            DataFileStream<SyslogRecord> reader = new DataFileStream<>(
+                    inputStream,
+                    new SpecificDatumReader<>(SyslogRecord.class)
+            );
+            SyslogRecord syslogRecord = null;
+            LOGGER.info("\nReading records from file {}:", hdfsreadpath);
+
+            Assertions.assertTrue(reader.hasNext());
```

Comment:
A for loop seems appropriate here.

Reply:
Solved in commit b0eb72b1b8609cf588a3f0f519765b3a06a365d7